Index: UF3 File management/Exercices/src/Ex_java/io/Ex3.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/UF3 File management/Exercices/src/Ex_java/io/Ex3.java b/UF3 File management/Exercices/src/Ex_java/io/Ex3.java
new file mode 100644
--- /dev/null	(date 1684003245629)
+++ b/UF3 File management/Exercices/src/Ex_java/io/Ex3.java	(date 1684003245629)
@@ -0,0 +1,44 @@
+package Ex_java.io;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+
+public class Ex3 {
+    public static void main(String[] args) {
+        File workspaceDir = new File(System.getProperty("user.dir"));
+        File fotosDir = new File(workspaceDir, "fotos");
+        if (!fotosDir.exists()) {
+            fotosDir.mkdir();
+        }
+
+        File descargasDir = new File(System.getProperty("user.home"), "descargas");
+        File[] pngFiles = descargasDir.listFiles((dir, name) -> name.toLowerCase().endsWith(".png"));
+        for (File pngFile : pngFiles) {
+            try {
+                Files.move(pngFile.toPath(), new File(fotosDir, pngFile.getName()).toPath(), StandardCopyOption.REPLACE_EXISTING);
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+
+        deleteDir(fotosDir);
+    }
+
+    private static void deleteDir(File dir) {
+        if (dir.exists()) {
+            File[] files = dir.listFiles();
+            if (files != null) {
+                for (File file : files) {
+                    if (file.isDirectory()) {
+                        deleteDir(file);
+                    } else {
+                        file.delete();
+                    }
+                }
+            }
+            dir.delete();
+        }
+    }
+}
Index: UF3 File management/Exercices/src/Ex_java/io/Ex2.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/UF3 File management/Exercices/src/Ex_java/io/Ex2.java b/UF3 File management/Exercices/src/Ex_java/io/Ex2.java
new file mode 100644
--- /dev/null	(date 1684002792683)
+++ b/UF3 File management/Exercices/src/Ex_java/io/Ex2.java	(date 1684002792683)
@@ -0,0 +1,38 @@
+package Ex_java.io;
+
+import com.sun.tools.javac.Main;
+
+import java.io.File;
+
+public class Ex2 {
+
+    public static void main(String[] args) {
+        Ex2 programa = new Ex2();
+        programa.inicio();
+    }
+
+    public void inicio() {
+        // Rutas absolutas
+        File directoryAbsPath = new File("c:/Temp");
+        File fileAbsPath = new File("c:/Temp/data.txt");
+
+        // Rutas relativas
+        File directoryRelativePath = new File("Temp");
+        File fileRelativePath = new File("Temp/data.txt");
+
+        // Mostramos información para cada ruta
+        showPaths(directoryAbsPath, false);
+        showPaths(fileAbsPath, false);
+        showPaths(directoryRelativePath, true);
+        showPaths(fileRelativePath, true);
+    }
+
+    public void showPaths(File f, boolean relativePath) {
+        System.out.println("**********************************");
+        if (relativePath) {
+            System.out.println("La ruta es " + f.getAbsolutePath());
+        }
+        System.out.println("Su padre es " + f.getParent());
+        System.out.println("Su nombre es " + f.getName());
+    }
+}
Index: UF3 File management/Exercices/numbers.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/UF3 File management/Exercices/numbers.txt b/UF3 File management/Exercices/numbers.txt
new file mode 100644
--- /dev/null	(date 1684003717643)
+++ b/UF3 File management/Exercices/numbers.txt	(date 1684003717643)
@@ -0,0 +1,1 @@
+1,2 2,4 3,6 4,7 5,8 6,1 10,3
\ No newline at end of file
Index: UF3 File management/Exercices/src/Ex_java/io/Ex1.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/UF3 File management/Exercices/src/Ex_java/io/Ex1.java b/UF3 File management/Exercices/src/Ex_java/io/Ex1.java
new file mode 100644
--- /dev/null	(date 1684002513588)
+++ b/UF3 File management/Exercices/src/Ex_java/io/Ex1.java	(date 1684002513588)
@@ -0,0 +1,33 @@
+package Ex_java.io;
+
+import com.sun.tools.javac.Main;
+
+import java.io.File;
+
+public class Ex1 {
+
+    public static void main(String[] args) {
+        inicio();
+    }
+
+    public static void inicio() {
+
+        File directoryAbsPath= new File ("c:/Temp");
+        File fileAbsPath = new File ("c:/Temp/data.txt");
+        File directoryRelativePath = new File("Temp");
+        File fileRelativePath = new File("Temp/data.txt");
+
+        showPaths(directoryAbsPath, false);
+        showPaths(fileAbsPath, false);
+        showPaths(directoryRelativePath, true);
+        showPaths(fileRelativePath, true);
+
+    }
+
+    public static void showPaths(File f, boolean relativePath) {
+        System.out.println("**********************************");
+        if (relativePath) System.out.println("La ruta es " + f.getAbsolutePath());
+        System.out.println("Su padre es " + f.getParent());
+        System.out.println("Su nombre es " + f.getName());
+    }
+}
